### DISCUSSION TITLE
Is there a way in MOOSE to perform convergence check on variables in Explicit solver

### DISCUSSION THREAD

**Discussion by chunhuizhao478** [idaholab/moose#26199] [Q&A General]
<img width="573" alt="Screenshot 2023-11-29 at 3 34 57 PM" src="https://github.com/idaholab/moose/assets/93891863/47ee8bad-3fc2-4167-bf2b-1b0ddf0a0713">

Hi all, I would like to solve poroelasticity equation using explicit solver, I acknowledge the existence of porous flow modulus but it's fully implicit, I want to estimate how far we could go in the explicit solver way.

As shown in the equations above, we solve displacement `u` from first equation 1, and plug into equation 2 for pressure `p`, until convergence is achieved if both `u` `p` relative error is below the threshold, we forbid the marching in time. I'm not sure if there is something in the `control system` for example we could utilize for this purpose. I want to know how feasible this approach is in MOOSE, thanks!

**Comment by GiudGiud** [comment]
Hello

To define arbirtary convergence criteria you can use this:
https://mooseframework.inl.gov/source/userobjects/Terminator.html

Otherwise we are going to be working on a new system, called Convergence, that will enable everything convergence related.
It s not ready yet

Guillaume

**Reply by chunhuizhao478** [reply]
Thanks @GiudGiud. I checked this function, it may look like what I want, however, I don't want to cut the time step by half (just iterative at current time step), but I want to save the results I have for this iteration before a restart so I could compare the relative error, is there a way to achieve this?

**Reply by GiudGiud** [reply]
This object can terminate the simulation normally. Cutting the timestep is not the only thing it can do

The docs mention how to do that iirc

**Reply by chunhuizhao478** [reply]
I understand if I set `fail_mode = HARD`, it will terminate the simulation, however I don't want the simulation ends (which is handled by setting `fail_mode = SOFT`, but also I want the time to stay at current value until some criterion is met, is that possible?

**Reply by GiudGiud** [reply]
ok so you want to call it convergence and accept the time step. Not sure we have that
